homeassistant/components/knx/sensor.py

```diff
@@ -226,3 +196,209 @@ async def async_added_to_hass(self) -> None:
                 self.after_update_callback,
             )
         )
+
+
+class KnxSensor(SensorEntity):
+    """Base class for a KNX sensor."""
+
+    _device: XknxSensor
+
+    @property
+    def native_value(self) -> StateType:
+        """Return the state of the sensor."""
+        return self._device.resolve_state()
+
+    @property
+    def extra_state_attributes(self) -> dict[str, Any] | None:
+        """Return device specific state attributes."""
+        if self._device.last_telegram is not None:
+            return {
+                ATTR_SOURCE: str(self._device.last_telegram.source_address),
+            }
+        return None
+
+
+class KnxUiSensor(KnxSensor, KnxUiEntity):
+    """Representation of a KNX sensor configured from UI with new description."""
+
+    _device: XknxSensor
+
+    def __init__(self, description: KnxSensorDescription) -> None:
+        """Initialize of a KNX sensor."""
+        self.entity_description: KnxSensorDescription = description
+        self._device = description.device
+
+        super().__init__(description.xknx_module, description.key, {})
+        self._attr_device_info = description.device_info
+
+
+class KnxYamlSensor(KnxSensor, KnxYamlEntity):
+    """Representation of a KNX sensor configured from YAML with new description."""
+
+    _device: XknxSensor
+
+    def __init__(self, description: KnxSensorDescription) -> None:
+        """Initialize of a KNX sensor."""
+
+        super().__init__(description.xknx_module, description.device)
+        self.entity_description: KnxSensorDescription = description
+        self._attr_unique_id = description.key
+
+
+@dataclass(frozen=True, kw_only=True)
+class KnxSensorDescription(SensorEntityDescription):
+    """Class describing KNX system sensor entities."""
+
+    device: XknxSensor
+    xknx_module: KNXModule
+    device_info: DeviceInfo | None = None
+
+
+class KnxSensorDescriptionFactory:
```

Comment:
since this only holds `@staticmethod`, it can be either omitted and moved to plain module functions, or if you prefer, moved to methods or classmethods of `KnxSensorDescription`